packages/react/src/transactions/publications/useCreateMirror.ts

```diff
@@ -240,11 +242,13 @@ export function useCreateMirror(): UseDeferredTask<
       'You must have a profile to create a mirror.',
     );
 
-    const request = createMirrorRequest({
-      signless: session.profile.signless,
-      sponsored: args.sponsored ?? true,
-      ...args,
-    });
+    const request = configureRequest(
+      createMirrorRequest({
+        signless: session.profile.signless,
+        sponsored: args.sponsored ?? true,
+        ...args,
+      }),
+    );
```

Comment:
Given we do have places where we do:
```
const { config } = useSharedDependencies();
```
wondering if it's more straightforward to also retrieve `config` in these hooks and make the fallback logic apparent here. Rather than creating an request object and rely on `configureRequest` to override it.

Reply:
in some places, with simple logic, it could be apparent, but in complicated cases where the sponsored flag is set based on multiple conditions, adding one extra condition to the mix will only make it less obvious, and more error prone, than just overwriting it at the end, if needed at all (if defined in the config).

Reply:
I added overwrites to all hooks that would require it, none uses `useSharedDependencies` beside `useCreatePostRequest` that has the most complicated logic and external calls to build the Request.

Reply:
Fair enough